src/dodal/devices/tetramm.py

```diff
@@ -214,11 +215,20 @@ def __init__(
         prefix: str,
         directory_provider: DirectoryProvider,
         name: str,
+        type: str | None = None,
         **scalar_sigs: str,
     ) -> None:
         self.drv = TetrammDriver(prefix + "DRV:")
         self.hdf = NDFileHDF(prefix + "HDF5:")
         controller = TetrammController(self.drv)
+        config_signals = [
+            self.drv.values_per_reading,
+            self.drv.averaging_time,
+            self.drv.sample_time,
+        ]
+        if type:
+            type_signal, _ = soft_signal_r_and_setter(str, type)
+            config_signals.append(type_signal)
```

Comment:
Following convention elsewhere in dodal:

```suggestion
        if type is not None:
            self.type, _ = soft_signal_r_and_setter(str, type)
        else:
            self.type = None
        config_signals = [
            self.drv.values_per_reading,
            self.drv.averaging_time,
            self.drv.sample_time,
        ]
```

Reply:
That doesn't get it into the config_signals list without then doing the same "if type:" again.

Reply:
Good point, you'll need to append to config signals